vncviewer/UserDialog.cxx

```diff
@@ -62,34 +63,54 @@ static void button_cb(Fl_Widget *widget, void *val) {
   widget->window()->hide();
 }
 
+static void keepPassword(Fl_Widget *widget, void *data) {
+  UserDialog *dialog = (UserDialog*) data;
+  Fl_Check_Button* flcb = (Fl_Check_Button*)widget;
+  dialog->setKeepPassword(flcb->value());
+}
+
 UserDialog::UserDialog()
 {
+  user_ = NULL;
+  password_ = NULL;
+  keepPasswd_ = true;
 }
 
 UserDialog::~UserDialog()
 {
+  resetPassword();
+}
+
+void UserDialog::resetPassword()
+{
+  if(user_) { free(user_); user_ = NULL; }
+  if(password_) { free(password_); password_ = NULL; }
```

Comment:
We don't need to save lines, so I'd prefer if this is more readable by having things on their own lines. So the same style as the rest of the code. :)